bookshelf/4-auth/pom.xml

```diff
@@ -206,6 +206,10 @@ Copyright 2016 Google Inc.
         <groupId>com.google.cloud.tools</groupId>
         <artifactId>appengine-maven-plugin</artifactId>
         <version>${appengine.maven.plugin}</version>
+        <configuration>
+          <projectId>${projectID}</projectId>
+          <version>GCLOUD_CONFIG</version> <!-- let gcloud generate a new version -->
```

Comment:
nit: this might not be descriptive enough. Maybe: "option allows for an autogenerated GAE version"